src/lib.rs

```diff
@@ -490,6 +490,47 @@ impl Client {
     }
 
     async fn post(
+        &self,
+        payload: Option<&impl Serialize>,
+        mut nonce: Option<String>,
+        signer: &impl Signer,
+        url: &str,
+    ) -> Result<BytesResponse, Error> {
+        let mut retries = 3;
+        loop {
+            let mut response = self
+                .post_attempt(payload, nonce.clone(), signer, url)
+                .await?;
+            if response.parts.status != StatusCode::BAD_REQUEST {
+                return Ok(response);
+            }
+            let body = response.body.into_bytes().await.map_err(Error::Other)?;
+            let problem = serde_json::from_slice::<Problem>(&body)?;
+            if let Some("urn:ietf:params:acme:error:badNonce") = problem.r#type.as_deref() {
+                retries -= 1;
+                if retries != 0 {
+                    // Retrieve the new nonce from the header. If it
+                    // isn't there (it should be, the spec requires
+                    // it) then we will manually refresh a new one in
+                    // `post_attempt` due to `nonce` being `None` but
+                    // using the header saves us a request.
+                    nonce = response
+                        .parts
+                        .headers
+                        .get("Replay-Nonce")
+                        .and_then(|v| v.to_str().map(|s| s.to_string()).ok());
```

Comment:
I think this is `nonce_from_response()`? Can we use that here?

Reply:
Yeah it is. Done!